code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,387 @@
 	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
 	RegisterSignal(src, COMSIG_PARENT_QDELETING, PROC_REF(alert_admins_on_destroy))
 
+///Mode Defines
+#define SECSWORD_OFF 0
+#define SECSWORD_STUN 1
+#define SECSWORD_BURN 2
+
+//Security Sword
+/obj/item/melee/secsword
+	name = "securiblade"
+	desc = "A simple, practical blade developed by Shellguard munitions for ‘enhanced’ riot control."
+	lefthand_file = 'icons/mob/inhands/weapons_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons_righthand.dmi'
+	icon_state = "secsword0"
+	item_state = "secsword0"
+	flags = CONDUCT
+	force = 15
+	throwforce = 5
+	w_class = WEIGHT_CLASS_BULKY
+	sharp = TRUE
+	origin_tech = "combat=4"
+	attack_verb = list("stabbed", "slashed", "sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	materials = list(MAT_METAL = 1000)
+	needs_permit = TRUE
+	///The icon the sword has when turned off
+	var/base_icon = "secsword"
+	///How much stamina damage the sword does in stamina mode
+	var/stam_damage = 35 // 3 hits to stamcrit
+	///How much burn damage the sword does in burn mode
+	var/burn_damage = 10
+	/// How much power does it cost to stun someone
+	var/stam_hitcost = 1000
+	/// How much power does it cost to burn someone
+	var/burn_hitcost = 1500
+	/// the initial cooldown tracks the time between stamina damage. tracks the world.time when the baton is usable again.
+	var/cooldown = 3.5 SECONDS
+	///The sword's power cell - starts high
+	var/obj/item/stock_parts/cell/high/cell
+	///The sword's current mode. Defaults to off.
+	var/state = SECSWORD_OFF
+
+/obj/item/melee/secsword/Initialize(mapload)
+	. = ..()
+	link_new_cell()
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 1, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
+
+/obj/item/melee/secsword/Destroy()
+	if(cell?.loc == src)
+		QDEL_NULL(cell)
```

Comment:
it should never be the case that the cell's loc is not `src` unless some fuckery is happening. I would just forgo this loc check and delete it, this won't cause problems as long as you're properly clearing references.